platform/javascript/js/libs/library_godot_display.js

```diff
@@ -73,15 +73,48 @@ const GodotDisplayVK = {
 			GodotDisplayVK.textarea = create('textarea');
 			GodotDisplayVK.updateSize();
 		},
-		show: function (text, multiline, start, end) {
+		show: function (text, type, start, end) {
 			if (!GodotDisplayVK.textinput || !GodotDisplayVK.textarea) {
 				return;
 			}
 			if (GodotDisplayVK.textinput.style.display !== '' || GodotDisplayVK.textarea.style.display !== '') {
 				GodotDisplayVK.hide();
 			}
 			GodotDisplayVK.updateSize();
-			const elem = multiline ? GodotDisplayVK.textarea : GodotDisplayVK.textinput;
+
+			let elem = GodotDisplayVK.textinput;
+			switch (type) {
+			case 0: // KEYBOARD_TYPE_DEFAULT
+				elem.inputmode = 'text';
+				break;
+			case 1: // KEYBOARD_TYPE_MULTILINE
+				elem = GodotDisplayVK.textarea;
+				break;
+			case 2: // KEYBOARD_TYPE_NUMBER
+				elem.inputmode = 'numeric';
+				break;
+			case 3: // KEYBOARD_TYPE_NUMBER_DECIMAL
+				elem.inputmode = 'decimal';
+				break;
+			case 4: // KEYBOARD_TYPE_PHONE
+				elem.inputmode = 'tel';
+				break;
+			case 5: // KEYBOARD_TYPE_EMAIL_ADDRESS
+				elem.inputmode = 'email';
+				break;
+			case 6: // KEYBOARD_TYPE_PASSWORD
+				// Password input mode is being considered in a future spec
+				// https://github.com/whatwg/html/issues/4875
+				elem.inputmode = 'text';
+				break;
+			case 7: // KEYBOARD_TYPE_URL
+				elem.inputmode = 'url';
+				break;
+			default:
+				elem.inputmode = 'text';
+				break;
+			}
```

Comment:
I'm a bit surprised by this.
HTML does have a standard `<input>` `type=` attribute for most of those cases:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-type

and that actually includes "password". The only missing one is `decimal` I guess. Any reason to use `inputmode` (which seems designed mostly for elements that do not support the `type` attribute like `input`)?

Reply:
Yeah in hindsight that looks like a pretty obvious way to do this. I can't remember why I used inputmode.

Reply:
For numeric types, from what I read it seems that using `type="text" inputmode="..."` is a better option. Using `type="number"` sanitizes `elem.value` to a floating point number (or empty) per the [HTML spec](https://html.spec.whatwg.org/multipage/input.html#number-state-(type=number)). We should leave number parsing/validation to Godot users if they want it.

For `type="url"` and `type="email"`, the browser is also supposed to sanitize `value` by stripping leading and trailing whitespace. This isn't as invasive as `type="number"` so I think it's okay to use?

I'll switch to using `type` for the rest.